aggregator/apps/api/v1/stitcher.py

```diff
@@ -244,11 +255,21 @@ def get_ballots_for_date(self, date):
                     ),
                     {},
                 )
+
+                # Look for existing cancellation metadata (e.g, we've added "cancelled_election" metadata
+                # in EE) and if it doesn't exist, add our own cancellation reason as a metadata object.
+                # We do this because this is the first time we know about the number of candidates, so we can
+                # infer the reason for the cancellation
                 wdiv_ballot["seats_contested"] = wcivf_ballot.get("seats_contested", 0)
                 wdiv_ballot["candidates"] = wcivf_ballot.get("candidates", [])
-                wdiv_ballot["cancellation_reason"] = get_ballot_cancellation_reason(
-                    wdiv_ballot
-                )
+                existing_metadata = wdiv_ballot["metadata"] or {}
```

Comment:
I'm not sure this will ever reach the "or" case, I think it'll throw an exception if the key doesn't exist. Maybe switch over to using `.get()` with a default of `{}`

Reply:
Yeah, I got confused by this!

So, `dict().get()` will return a thing _if the key exists_, but it doesn't care if that key is `None`. So if you always want a dict, the following wont work:

```py
foo= {"bar": None}
assert foo.get("bar", {}) == {}
```

However `assert foo["bar"] or  {} == {}` does work.

`metadata` is always a key in the response, so this ends up working. We could deal with the case where `metadata` _isn't_ always there, but I think we make a load of assumptions about the shape of the dict already, elsewhere.

Reply:
Oh I see, I thought this was a check if the key was present rather than a check on its value. Sorry!